src/app/members/components/MemberHeader.tsx

```diff
@@ -31,12 +31,13 @@ export const MemberHeader = () => {
   };
 
   useEffect(() => {
-    if (debounceValue.length) {
+    const shouldNavigateToMembers = pathname === Routes.Members;
+    if (debounceValue) {
       router.push(`${Routes.Members}?key=${debounceValue}`);
-    } else {
+    } else if (shouldNavigateToMembers) {
```

Comment:
not a good approach, we can improve

Reply:
@muhammad-bilal-ali-khokhar  I have used better approach.